server/src/main/java/org/apache/druid/server/security/AuthConfig.java

```diff
@@ -107,6 +144,36 @@ public boolean authorizeQueryContextParams()
     return authorizeQueryContextParams;
   }
 
+  /**
+   * Filter the user-supplied context keys based on the context key security
+   * rules. If context key security is disabled, then allow all keys. Else,
+   * apply the three key lists defined here.
+   * <ul>
+   * <li>Allow Druid-defined keys.</li>
+   * <li>Allow anything not in the secured context key list.</li>
+   * <li>Allow anything in the config-defined unsecured key list.</li>
+   * </ul>
+   * In the typical case, a site defines either the secured key list
+   * (to handle a few keys that are <i>are not</i> allowed) or the unsecured key
+   * list (to enumerate a few that <i>are</i> allowed.) If both lists
+   * are given, think of the secured list as exceptions to the unsecured
+   * key list.
+   *
+   * @return the list of secured keys to check via authentication
+   */
+  public Set<String> filterContextKeys(final Set<String> userKeys)
```

Comment:
nit - could we call this method `contextKeysToAuthorize` or `filterContextKeysToAuthorize`?